gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -210,13 +253,15 @@ static __always_inline void trace_v4(struct pt_regs *ctx, __u64 pid_tgid,
 	event->tid = (u32)pid_tgid;
 	event->uid = (u32)uid_gid;
 	event->gid = (u32)(uid_gid >> 32);
-	event->src.version = event->dst.version = 4;
-	event->src.proto_raw = event->dst.proto_raw = IPPROTO_TCP;
-	BPF_CORE_READ_INTO(&event->src.addr_raw.v4, sk,
-			   __sk_common.skc_rcv_saddr);
-	BPF_CORE_READ_INTO(&event->dst.addr_raw.v4, sk, __sk_common.skc_daddr);
-	event->dst.port = dport;
-	event->src.port = BPF_CORE_READ(sk, __sk_common.skc_num);
+	if (src_dst_entry) {
+		event->src = src_dst_entry->src;
+		event->dst = src_dst_entry->dst;
+	} else {
+		// src_dst_entry is not set when tcp_v{4|6}_connect is not called, e.g.,
+		// when calling inet_stream_connect on an already connected socket. So,
+		// try to read the endpoints from the socket here.
+		read_l4endpoints_from_sock_v4(&event->src, &event->dst, sk);
```

Comment:
The logic does not work in some edge-cases. But it was already not working before.

Steps to reproduce:

* Start the gadget:
```
$ sudo -E ig run ghcr.io/inspektor-gadget/gadget/trace_tcpconnect:burak-trace_tcpconnect-fix --verify-image=false
WARN[0000] image signature verification is disabled due to using corresponding option 
WARN[0000] image signature verification is disabled due to using corresponding option 
RUNTIME.CONTAINERNAME                 SRC                                               DST                                              COMM                    PID        TID ERROR         
```

* Start a container. To make it reproduce faster, change the tcp timeout of the container to approximately 7 seconds:
```
$ docker run -ti --rm --privileged busybox
/ # sysctl net.ipv4.tcp_syn_retries=2
net.ipv4.tcp_syn_retries = 2
```

* Start a tcp connection to nowhere and quickly stop the process with ctrl-z (SIGSTOP), and resume it with `fg`:
```
/ # wget 1.2.3.4
Connecting to 1.2.3.4 (1.2.3.4:80)
^Z[1]+  Stopped                    wget 1.2.3.4
/ # fg
```

* Notice the gadget output after waiting a few seconds more:
```
RUNTIME.CONTAINERNAME  SRC               DST         COMM  PID     TID ERROR
boring_edison          172.17.0.3:32798  1.2.3.4:80  wget  434444  434444 error#512
boring_edison          0.0.0.0:0         1.2.3.4:0   wget  434444  434444 ETIMEDOUT
```

There is two events (one emitted at the ctrl-z, and one emitted at the end of the tcp timeout).